src/uct/ib/mlx5/dv/ib_mlx5dv_md.c

```diff
@@ -731,8 +733,16 @@ uct_ib_mlx5_devx_reg_mr(uct_ib_mlx5_md_t *md, uct_ib_mlx5_devx_mem_t *memh,
         /* Fallback if multi-thread registration is unsupported */
     }
 
-    status = uct_ib_reg_mr(&md->super, address, length, params, access_flags,
-                           NULL, &memh->mrs[mr_type].super.ib);
+    do {
+        status = uct_ib_reg_mr(&md->super, address, length, params, access_flags,
+                               NULL, &memh->mrs[mr_type].super.ib,
+                               attempt_cnt == 0);
+        attempt_cnt++;
+        access_flags = uct_ib_memh_access_flags(&memh->super,
```

Comment:
First `reg_mr`, then `access_flags` - is that intentional?
This line always has `attempt_cnt` >= 1...

Reply:
Yes, it's intentional, though it might be written better..
`access_flags` is first calculated at the top of the function because it's used for another function, and only then it is used in the `do while` loop